queries/schedules.py

```diff
@@ -90,271 +95,831 @@ def help_text(lemma: str) -> str:
-    "dagskráin" | "sjónvarpsdagskráin"
+QSchAM →
+    "fyrir" "hádegi"
 
-QSchThisEvening →
-    "núna"? "í_kvöld"
+QSchPM →
+    "eftir" "hádegi"
 
 $score(+55) QSchedule
 
 """
 
 
-def QSchTelevisionQuery(node, params, result):
+def QScheduleQuery(node: Node, params: ParamList, result: Result) -> None:
     result.qtype = _SCHEDULES_QTYPE
-    result.qkey = _TELEVISION_QKEY
 
 
-def QSchTelevisionEveningQuery(node, params, result):
-    result.qtype = _SCHEDULES_QTYPE
-    result.qkey = _TELEVISION_EVENING_QKEY
+def QSchWill(node: Node, params: ParamList, result: Result) -> None:
+    result["will_be"] = True
 
 
-def QSchRadioStationNowQuery(node, params, result):
-    result.qtype = _SCHEDULES_QTYPE
-    result.qkey = _RADIO_QKEY
+def QSchPM(node: Node, params: ParamList, result: Result) -> None:
+    result["PM"] = True
 
 
-def _clean_desc(d: str) -> str:
-    """ Return first sentence in multi-sentence string. """
-    return d.replace("Dr.", "Doktor").replace("?", ".").split(".")[0]
+def QSchAM(node: Node, params: ParamList, result: Result) -> None:
+    result["PM"] = False
 
 
-_RUV_TV_SCHEDULE_API_ENDPOINT = "https://apis.is/tv/ruv/"
-_TV_API_ERRMSG = "Ekki tókst að sækja sjónvarpsdagskrá."
-_CACHED_TV_SCHEDULE: Optional[List] = None
-_TV_LAST_FETCHED: Optional[datetime] = None
+def QSchSérnafn(node: Node, params: ParamList, result: Result) -> None:
+    channel = result._nominative.replace("Stöðvar", "Stöð")
 
+    if channel == "Stöð 2 Sport 2":
+        QSchStod2Sport2(node, params, result)
+    elif channel == "Stöð 2 Sport":
+        QSchStod2Sport(node, params, result)
+    elif channel == "Stöð 2 Bíó":
+        QSchStod2Bio(node, params, result)
+    elif channel == "Stöð 2":
+        QSchStod2(node, params, result)
 
-def _query_tv_schedule_api() -> Optional[List]:
-    """ Fetch current television schedule from API, or return cached copy. """
-    global _CACHED_TV_SCHEDULE
-    global _TV_LAST_FETCHED
-    if (
-        not _CACHED_TV_SCHEDULE
-        or not _TV_LAST_FETCHED
-        or _TV_LAST_FETCHED.date() != datetime.today().date()
-    ):
-        # Not cached. Fetch data.
-        _CACHED_TV_SCHEDULE = None
-        sched = query_json_api(_RUV_TV_SCHEDULE_API_ENDPOINT)
-        if isinstance(sched, dict) and "results" in sched and len(sched["results"]):
-            _TV_LAST_FETCHED = datetime.utcnow()
-            _CACHED_TV_SCHEDULE = sched["results"]
-    return _CACHED_TV_SCHEDULE
 
+def QSchRUV(node: Node, params: ParamList, result: Result) -> None:
+    result["api_channel"] = "ruv"
+    result["channel"] = "RÚV"
+    result["channel_type"] = "tv"
+    result["station"] = "RÚV"
 
-_RUV_RADIO_SCHEDULE_API_ENDPOINT = "https://muninn.ruv.is/files/xml/{0}/{1}/"
-_RADIO_API_ERRMSG = "Ekki tókst að sækja útvarpsdagskrá."
-_RADIO_SCHED_CACHE: Dict[str, Any] = {}
-_RADIO_LAST_FETCHED: Dict[str, datetime] = {}
 
+def QSchRUV2(node: Node, params: ParamList, result: Result) -> None:
+    result["api_channel"] = "ruv2"
```

Comment:
Maybe write separate utility function to add these attributes to the Result object to reduce code?

Reply:
Good point, added a helper function which cleans up the code.